dsl/src/test/java/org/mule/test/config/ast/ArtifactAstValidationsTestCase.java

```diff
@@ -53,6 +84,65 @@
 @Story(DSL_VALIDATION_STORY)
 public class ArtifactAstValidationsTestCase extends AbstractMuleContextTestCase {
 
+  private Injector injector;
+
+  @Rule
+  public SystemProperty systemProperty = new SystemProperty(ENFORCE_EXPRESSION_VALIDATION_PROPERTY, "true");
+
+  class BasicModule extends AbstractModule {
+
+    private Map<org.mule.runtime.api.config.Feature, Boolean> featureBooleanMap =
+        singletonMap(ENFORCE_EXPRESSION_VALIDATION, Boolean.valueOf(true));
+
+    @Override
+    protected void configure() {
+      bind(ArtifactAstValidatorBuilder.class).to(DefaultValidatorBuilder.class);
+      bind(ExtendedExpressionManager.class).to(DefaultExpressionManager.class);
+      bind(ValidationsProvider.class).to(CoreValidationsProvider.class);
+      bind(MuleContext.class).to(DefaultMuleContext.class);
+      bind(TransformersRegistry.class).to(DefaultTransformersRegistry.class);
+      bind(Object.class).to(String.class);
+      bind(Transformer.class).to(StringToBoolean.class);
+    }
+
+    @Provides
+    @Singleton
+    public ExpressionLanguage provideExpressionLanguage() {
+      return new WeaveDefaultExpressionLanguageFactoryService(null).create();
+    }
+
+    @Provides
+    @Singleton
+    public FeatureFlaggingService provideFeatureFlaggingService() {
+      return new DefaultFeatureFlaggingService("abcd", featureBooleanMap);
```

Comment:
what do these parameters mean?

Reply:
We are just trying to verify all injections work and in valid DataWeave expression is validated, so we need to set ENFORCE_EXPRESSION_VALIDATION to true as it defaults to false, passing it via a map, artifact id here does not matter

Reply:
FeatureFlaggingService needs to be injected, but tryting just give it empty HashMap so that we dont set ENFORCE_EXPRESSION_VALIDATION to true, that fails the test as defaullts to false